### PR TITLE
[Feature] Implementing a `find_lr` method

### PR DESCRIPTION
**Pull request by dcfidalgo**
This PR introduces a new `Pipeline.find_lr()` method and closes issue #183 . The new method is based on AllenNLP's `search_learning_rate` function and just returns two lists: learning rates and losses, the user is responsible for plotting the data or apply some smoothing first.

Apart from the `find_lr()` method, this PR also includes some refactoring to reuse code, as well as exposing the `no_grad` option in the `TrainerConfiguration`.

### CODE REVIEW

File: src/biome/text/pipeline.py [hunk 1]
```diff
@@ -154,6 +156,64 @@ def init_prediction_cache(self, max_size: int) -> None:
...
+            A configuration for finding the learning rate
+        training_data
+            The training data
+
+        Returns
+        -------
+        (learning_rates, losses)
+            Returns list of learning rates and corresponding losses.
+            Note: The losses are recorded before applying the corresponding learning rate
+        """
```
Comment (frascuchon) [inline src/biome/text/pipeline.py:183]: It would be nice include a minimal code snippet of how to use this method for learning rate selection (if possible)

Comment (dcfidalgo) [reply src/biome/text/pipeline.py:183]: Sorry, i totally forgot about this comment!
I planed to include a more detailed usage in one or two of our tutorials. There are a lot of blog posts/publications on how to select the learning rate once you made a learning rate scan, in my experience this learning rate scan gives you a rough estimate where to look for the learning rate, but to find the optimal one you still have to do a bit of HPO. I will include a link in the docstring pointing to one of those blog posts.

File: src/biome/text/pipeline.py [hunk 2]
```diff
@@ -154,6 +156,66 @@ def init_prediction_cache(self, max_size: int) -> None:
...
+            A trainer configuration
+        find_lr_config
+            A configuration for finding the learning rate
+        training_data
+            The training data
+
+        Returns
+        -------
+        (learning_rates, losses)
+            Returns list of learning rates and corresponding losses.
```
Comment (dcfidalgo) [inline src/biome/text/pipeline.py:183]:
```suggestion
            Returns a list of learning rates and corresponding losses.
```
